src/documents/barcodes.py

```diff
@@ -231,13 +233,40 @@ def detect(self) -> None:
             logger.debug("Scanning for barcodes using ZXING")
 
         try:
-            pages_from_path = convert_from_path(
-                self.pdf_file,
-                dpi=settings.CONSUMER_BARCODE_DPI,
-                output_folder=self.temp_dir.name,
+            # Read number of pages from pdf
+            num_of_pages = pdfinfo_from_path(self.pdf_file)["Pages"]
+            logger.debug(f"PDF has {num_of_pages} pages")
+
+            # Get limit from configuration
+ barcode_max_pages = (
+                num_of_pages
+                if settings.CONSUMER_BARCODE_MAX_PAGES == 0
+                else settings.CONSUMER_BARCODE_MAX_PAGES
             )
 
-            for current_page_number, page in enumerate(pages_from_path):
+            if barcode_max_pages < num_of_pages:
```

Comment:
```suggestion
            if barcode_max_pages < num_of_pages:  # pragma: no cover
```

This would be fine for the log line